container.php

```diff
@@ -196,7 +196,18 @@ public function onRoute(ActionEvent $actionEvent)
             $buildings = $container->get(BuildingRepositoryInterface::class);
 
             return function (Command\RegisterNewBuilding $command) use ($buildings) {
-                $buildings->add(Building::new($command->name()));
+                $buildings->store(Building::new($command->name()));
+            };
+        },
+        Command\CheckInUser::class => function (ContainerInterface $container) : callable {
+            $buildings = $container->get(BuildingRepositoryInterface::class);
+
+            return function (Command\CheckInUser $command) use ($buildings) {
+                $building = $buildings->get($command->buildingId());
+
+                $building->checkInUser($command->username());
```

Comment:
Note that the building is already in the DB (it is a sequence of events in the event store), so we do not create a new instance here, but rather work with the existing instance.